lib/koans/10_processes.ex

```diff
@@ -26,15 +45,71 @@ defmodule Processes do
   end
 
   koan "A common pattern is to include the sender in the message, so that it can reply" do
-    pid = spawn(fn -> receive do
-                         {:hello, sender} -> send sender, :how_are_you?
-                      end
-                 end)
+    greeter = fn ->
+      receive do
+        {:hello, sender} -> send sender, :how_are_you?
+      end
+    end
+
+    pid = spawn(greeter)
 
     send pid, {:hello, self()}
     assert_receive ___
   end
 
+  def yelling_echo_loop do
+    receive do
+      {caller, value} ->
+        send caller, String.upcase(value)
+        yelling_echo_loop
+    end
+  end
+
+  koan "Use tail recursion (calling a function as the very last statement) to receive multiple messages" do
+    pid = spawn &yelling_echo_loop/0
+
+    send pid, {self, "o"}
+    send pid, {self, "hai"}
+
+    receive do
+      msg -> assert msg == ___
+    end
+
+    receive do
+      msg -> assert msg == ___
+    end
+
+    Process.exit(pid, :kill)
```

Comment:
Not sure about that line but I'm also afraid of leaking processes. I know that it used to be a problem but I'm not sure if it still is. 

In theory since all koans run in their own process, I'd expect them to tear down everything they created once the koan exits. I don't know if this is how it currently works though because I don't fully comprehend the linking mechanism and which processes erlang decides to kill/keep alive.

I'd propose to keep this line in but investigate the overall issue. End game is to get to a point where koans can start child processes that get automatically cleaned up when the koan process exits. Shouldn't affect this PR though.